### ISSUE TITLE
Add support for non-readonly mounts from config.

### ISSUE THREAD

**Issue by Dirbaio** [kubernetes/kubeadm#628]
It should be possible to mount a directory read/write in `apiServerExtraVolumes` & co.

For example, to export apiserver audit logs like this:

```yaml
apiVersion: kubeadm.k8s.io/v1alpha1
kind: MasterConfiguration
(...)
apiServerExtraArgs:
  (...)
  audit-policy-file: /etc/kubernetes/audit-policy/audit-policy.yaml
  audit-log-path: /var/log/kube-audit/kube-audit.log
apiServerExtraVolumes:
  - name: audit-logs
    hostPath: /var/log/kube-audit
    mountPath: /var/log/kube-audit
```

This causes the volume to have `readOnly: true` in the apiserver manifest, so the apiserver can't write the logs.

There should be a way to set `readOnly: false` in the kubeadm config.

**Comment by timothysc** [comment]
TODO: cross link the PR that already addresses this.

**Comment by detiber** [comment]
This should be resolved with the merging of https://github.com/kubernetes/kubernetes/pull/60428 and available in v1.11
